### PR TITLE
Modify health check extension to add new metrics health check functions

### PR DESCRIPTION
**Pull request by skyduo**
**Description:** 
This is a new health check method which allow customers to monitor the health status of containers based on the exporter functionally failures. It will return a 200/500 status when call the endpoint based on if the components can send data to the destination properly. Currently it only supports monitoring the exporter health status, will also support receivers and processors in the future.

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector/issues/2573

**Testing:** 
`make otelcontribcol` and run executable file

**Documentation:** 
https://docs.google.com/document/d/1SpUMsWA2DeaoVazeQ8uEc1Wvu5LphmQU_TjzLmuJ4QM/edit#heading=h.rs1luwizct2w

original pr was creating a new health check extension, but some community members think we should add the new feature in the existing one. Had discussion with @bogdandrutu and @Aneurysm9 and create this new pr.
previous pr:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5144

### CODE REVIEW

File: extension/healthcheckextension/README.md
Comment (anuraaga) [inline extension/healthcheckextension/README.md:31]:
```suggestion
      enabled: true
```

To make it a valid example

File: extension/healthcheckextension/README.md
Comment (anuraaga) [inline extension/healthcheckextension/README.md:30]: This seems like it is health checking metrics, not basing a health check on metrics, to me. Naming is hard, but `check_export_pipeline` comes to mind to me

File: extension/healthcheckextension/config.go
Comment (anuraaga) [inline extension/healthcheckextension/config.go:71]: I believe this can be private

Comment (anuraaga) [reply extension/healthcheckextension/config.go:71]: Actually we probably don't need a separate method, just add it to the existing `createDefaultConfig` method I think. Or otherwise, let's  move this method to `factory.go` so the defaults are together

File: extension/healthcheckextension/healthcheckextension.go
Comment (anuraaga) [inline extension/healthcheckextension/healthcheckextension.go:80]: This approach doesn't seem scalable to more health check conditions - we should instead just have one, our current handler, and it can go through optional healthcheckers that report a health status. It would return an OK response only if all of these healthcheckers reported healthy.

@jpkrohling You had some thoughts on this, can you provide some guidance on how to structure this addition to the healthcheck?

File: extension/healthcheckextension/README.md
Comment (skyduo) [reply extension/healthcheckextension/README.md:31]: got it

File: extension/healthcheckextension/README.md
Comment (skyduo) [reply extension/healthcheckextension/README.md:30]: make sense, will change it.

File: extension/healthcheckextension/healthcheckextension.go
Comment (skyduo) [reply extension/healthcheckextension/healthcheckextension.go:80]: actually the current handler is defined in another jaeger package which i cannot change it, that's why i can only create my own handler.

File: extension/healthcheckextension/config.go
Comment (skyduo) [reply extension/healthcheckextension/config.go:71]: got it, will move to factory.

File: extension/healthcheckextension/README.md
Comment (skyduo) [reply extension/healthcheckextension/README.md:30]: this function only support exporter health check now but it will support receiver and processor in the future, do you think we can hard code a word `export` here?

File: extension/healthcheckextension/healthcheckextension.go
Comment (jpkrohling) [reply extension/healthcheckextension/healthcheckextension.go:80]: My idea was to move away from the Jaeger health check in favor of a proper lib for this. I haven't used, but one that I would give a try is this one: https://github.com/hellofresh/health-go

Comment (skyduo) [reply extension/healthcheckextension/healthcheckextension.go:80]: Another question is that can we make this an out of scope goal of this pr? we can have another further modification of the health check functions.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by skyduo** [comment]
not sure why `make golint` not work on my workspace even after i `make install-tools` and download the golint, but just resolve this issue it indicate

**Review by hossain-rayhan — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Comment by skyduo** [comment]
hi @bogdandrutu ! could you help check and approve this pr if possible? Anthony and Juraci has approved, just wait for your final approval.

**Review by jpkrohling — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

**Comment by skyduo** [comment]
hi @bogdandrutu! i just rebase and resolve the conflict, all tests pass now, could you help approve and merge it today?

**Pull request merged**
